services/system/AccountSys/src/AccountSys.cpp

```diff
@@ -55,20 +55,26 @@ namespace SystemService
       check(status.has_value(), "not started");
 
       auto sender = getSender();
-      check(sender == service || sender == inviteService, "Unauthorized account creation");
+      check(sender == service || sender == inviteService, "unauthorized account creation");
 
+      std::string strName = name.str();
       if (enable_print)
       {
          writeConsole("new acc: ");
-         writeConsole(name.str());
+         writeConsole(strName);
          writeConsole("auth con: ");
          writeConsole(authService.str());
       }
 
       check(name.value, "invalid account name");
+      check(strName.back() != '-', "account name must not end in a hyphen");
+      if (strName.size() > 2 && sender != service)
+      {
+         check(!strName.starts_with("x-"), "'x-' is a reserved prefix");
```

Comment:
Maybe add who it is reserved for, for clarity.

Reply:
As in, for subjective services? Technically it could really be for anything agreed upon by prods, though.